client/src/components/players/PlyrPlayer.vue

```diff
@@ -13,6 +13,22 @@ import "plyr/src/sass/plyr.scss";
 import type { MediaPlayerWithCaptions, MediaPlayerWithPlaybackRate } from "../composables";
 import { useCaptions } from "../composables";
 
+//helper type to avoid any in event payploads, maybe it can be better done.
+type HlsLevelLike = {
+	bitrate?: number;
+	width?: number;
+	height?: number;
+	name?: string;
+};
+type HlsManifestParsedData = { levels?: HlsLevelLike[] };
+type HlsLevelSwitchData = { level: number };
+
+type DashQualityChangeEvent = {
+	mediaType?: string;
+	newQuality: number;
+	reason?: unknown;
+};
+
```

Comment:
Do the hls and dash packages have `@types/...` type definition packages? that would be ideal